ddl/ddl_worker.go

```diff
@@ -91,6 +91,29 @@ func (d *ddl) isOwner() bool {
 	return isOwner
 }
 
+// buildJobDependence sets the curjob's dependency-ID.
+// The dependency-job's ID must less than the current job's ID, and we need the largest one in the list.
+func buildJobDependence(t *meta.Meta, curJob *model.Job) error {
+	jobs, err := t.GetAllDDLJobs()
```

Comment:
Should we run this check in a transaction? For example, two DDLs come in the same time. Will this function detect the dependence?

Reply:
It's in a transaction. The `t` has a field of transaction.